vertx-web/src/main/java/io/vertx/ext/web/sstore/AbstractSession.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.sstore;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.PRNG;
+import io.vertx.ext.web.Session;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author <a href="mailto:plopes@redhat.com">Paulo Lopes</a>
+ */
+public abstract class AbstractSession implements Session {
```

Comment:
If `AbstractSession` is to be used as a base for different implementations, could you document what it does and what an implementor is expected to code?